sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -293,15 +293,36 @@ Ext.define('SlateTasksTeacher.controller.Dashboard', {
 
         studentTask.set('TaskStatus', status);
 
-        me.doSaveStudentTask(studentTask, function(rec) {
-            taskRater.close();
-            me.getStudentTasksStore().load({
-                id: studentTask.getId(),
-                addRecords: true
-            });
+        me.doSaveStudentTask(studentTask, function(rec, success, request) {
+            var response = JSON.parse(request.getResponse().responseText),
+                validationErrors, key,
+                message = [
+                    'Unable to accept task.<br>'
+                ];
+
+            if (success) {
+                taskRater.close();
+                me.getStudentTasksStore().load({
+                    id: studentTask.getId(),
+                    addRecords: true
+                });
 
-            if (status === 're-assigned') {
-                me.doRateStudentTask(rec);
+                if (status === 're-assigned') {
+                    me.doRateStudentTask(rec);
+                }
+            } else {
+                rec.reject();
+                if (response.failed.length) {
```

Comment:
@nbey it is unsafe to assume that `response.failed` is defined, a connection failure could cause the request to fail with no response data. This should be `response.failed && response.failed.length`